source/Calamari.Common/Features/Processes/Semaphores/SystemSemaphoreManager.cs

```diff
@@ -5,24 +5,37 @@
 using Calamari.Common.Plumbing;
 using Calamari.Common.Plumbing.Extensions;
 using Calamari.Common.Plumbing.Logging;
+using Polly;
+using Polly.Retry;
 
 namespace Calamari.Common.Features.Processes.Semaphores
 {
     public class SystemSemaphoreManager : ISemaphoreFactory
     {
         readonly ILog log;
         readonly int initialWaitBeforeShowingLogMessage;
+        readonly ResiliencePipeline semaphoreAcquisitionPipeline;
 
         public SystemSemaphoreManager()
         {
             log = ConsoleLog.Instance;
             initialWaitBeforeShowingLogMessage = (int)TimeSpan.FromSeconds(3).TotalMilliseconds;
-        }
 
-        public SystemSemaphoreManager(ILog log, TimeSpan initialWaitBeforeShowingLogMessage)
-        {
-            this.log = log;
-            this.initialWaitBeforeShowingLogMessage = (int)initialWaitBeforeShowingLogMessage.TotalMilliseconds;
+            semaphoreAcquisitionPipeline = new ResiliencePipelineBuilder()
```

Comment:
Uses Polly's retry logic to do it for us

Reply:
I'm not sure what the consequences of not setting access control is, we'll either end up:

- All retries fail, there's no access control, we wait 500ms <-- **how likely is this to occur?**
- Succeeds on 1st/2nd try with or without access control, we wait 100ms

Are we better off with a shorter overall delay by performing less retries?

Reply:
Honestly I don't know how likely it is because I've not been able to reliably get it to fail.

I don't mind changing the delay to 50ms or drop the retry count to 3? If you considering a 500ms delay to be too long. My feeling in the context of Calamari starting up, that 500ms isn't the worst if it eventually fails anyway

Reply:
Changed to 50ms delay in [74552d6](https://github.com/OctopusDeploy/Calamari/pull/1414/commits/74552d682b016308928da243ef38be7db3c15c42)